Source/JavaScriptCore/assembler/MacroAssemblerARM64.h

```diff
@@ -276,11 +276,23 @@ class MacroAssemblerARM64 : public AbstractMacroAssembler<Assembler> {
             m_assembler.add<64>(dest, dest, UInt12(static_cast<int32_t>(immediate)));
             return;
         }
-        if (isUInt12(-immediate)) {
-            m_assembler.sub<64>(dest, dest, UInt12(static_cast<int32_t>(-immediate)));
+        if (isUInt12(toTwosComplement(immediate))) {
```

Comment:
nit: can we just make `add64(TrustedImm64 imm, RegisterID dest)` call `add64(TrustedImm64 imm, RegisterID src, RegisterID dest)` with `dest` passed to it as `src`?  Then, we can avoid so much replication of the same code.

Reply:
Changed.